### PR TITLE
Fix iTerm2 startup issues and additional options

### PR DESCRIPTION
**Pull request by crowedavid**
Building off of sammy's patch, this commit fixes script to startup iTerm2 and operate more reliably in a variety of iTerm2 configurations and if it isn't already running.

While in the code I also added two more property options:
- always open a new session/tab if Alfred starts iTerm2
- whether to raise and switch keyboard focus to show output if iTerm2 is already running

Updated README.md to describe optional property settings.

### CODE REVIEW

**Comment by natterstefan** [comment]
Will this resolve https://github.com/vitorgalvao/custom-alfred-iterm-scripts/issues/43 @crowedavid? Hopefully, this is merged soon :)

File: custom_iterm_script.applescript [hunk 1]
```diff
@@ -1,16 +1,41 @@
...
+			end if
+		end if
+		
 		delay 0.1 -- If we do not wait, the command may fail to send
-
-		tell the first window to tell current session to write text q
-		activate
+		
+		tell the current window to tell current session to write text q
+		if always_activate is true then
```
Comment (vitorgalvao) [inline custom_iterm_script.applescript:37]: Needs newline above.

File: README.md [hunk 2]
```diff
@@ -11,6 +11,12 @@ AppleScript for [iTerm2](https://iterm2.com/) [Alfred](https://www.alfredapp.com
 4. Select the text in the box.
 4. Paste.
 
+## Optional property settings:
+
+1. Set `open_in_new_window` to `true` to always open command in new window.
```
Comment (vitorgalvao) [inline README.md:16]: Please make the numbers correct (I know it’s only aesthetic for the Markdown and that above they’re not, those should be fixed as well).

**Comment by vitorgalvao** [comment]
I’ve just released a new version, and ultimately I’m deciding against this.

While I’ve added the option to open in a new window (from the other PR), opening a new session seems useless in that context (it’s either one or the other), and not bringing to front is a bad experience, because the only time it can work is when a window is already open and the user sets to open in a new tab.

**Comment by vitorgalvao** [comment]
I’ve just released a new version, and ultimately I’m deciding against this.

While I’ve added the option to open in a new window (from the other PR), opening a new session seems useless in that context (it’s either one or the other), and not bringing to front is a bad experience, because the only time it can work is when a window is already open and the user sets to open in a new tab.

Thank you anyway for the PR.

**Comment by crowedavid** [comment]
> I’ve just released a new version, and ultimately I’m deciding against this.

completely understand.

> Thank you anyway for the PR.

thanks for the colorful feedback on my coding style. it appears you have included the logic and actions using the nested "if/else/end" even though you didn't like them; they do resolve the startup and operational problems others have noted. the optional properties i included were only to offer a mechanism to enable behaviors i found useful.

i haven't tested your new release but will soon and comment on that thread.

**Comment by crowedavid** [comment]
> Will this resolve #43 @crowedavid? Hopefully, this is merged soon :)

my updated code did resolve #43. vitor's new release includes the logic and is on the master branch so i'd use it instead of mine.  i haven't tested it yet myself yet however.

**Comment by vitorgalvao** [comment]
> thanks for the colorful feedback on my coding style.

I’m unsure if you’ve taken my review badly. If you did, know that was not my intention.

I’m responsible for the code in this repo, which means that long after a PR is merged and the contributor is gone, I’m the one that has to maintain the code and keep it working. There was no criticism on your coding style—you can code however you want—but inside an existing project, consistency with existing code is important. In some cases you didn’t have examples to latch on to, so I needed to make the points now.

That’s what code reviews are for.

> it appears you have included the logic and actions using the nested "if/else/end" even though you didn't like them

I didn’t say I didn’t like them, I said I didn’t get the point because in your case it seemed to always lead to the same result, which defeated the purpose of a conditional. I now see one of the cases I wasn’t thinking about, though I haven’t tested.

Your PR did something important for the new release. It and the previous one and all the other code floating around showed how confusing it can be to deal with iTerm in AppleScript. It’s too verbose and conditionals are hard to parse mentally. That’s why I decided to define the handlers, so it can be extended more easily and we can keep the conditional logic short.

**Comment by crowedavid** [comment]
review not taken badly - just amused to see the running commentary.  your repo is your repo and i have no bones to pick there. eventually you'll leave as well.

i was solving problems i had with your previous code. it was primarily for myself but i put it up in hopes it would help someone else as well. i am glad it eventually led to an update others might find useful too.

i do like the handlers and will be testing your new code to see if it works as i need.

**Comment by vitorgalvao** [comment]
> just amused to see the running commentary.

Again, not supposed to be commentary, it’s a review.

If you check my profile, you might notice a lot of my time is spent on reviews and fixes, mostly maintaining [Homebrew Cask](https://github.com/Homebrew/homebrew-cask). It’s an important part of the process. It’s also one that takes long, so it’s done when improvement is expected. If I didn’t care for and appreciate your contribution, I wouldn’t have bothered.

> eventually you'll leave as well.

True, even if by death. But the important distinction is that while this is connected to my account, any issues that need solving will fall on my plate and time, not the ones of previous contributors, so I need to ensure contributions are things I can stand behind.
